rust-coroutine/src/main.rs

```diff
@@ -0,0 +1,35 @@
+extern crate coroutine;
+
+use coroutine::asymmetric::Coroutine;
+
+fn skynet(me: coroutine::asymmetric::CoroutineRef<(u64, u64)>) {
+    let (my_number, remaining) = me.take_data().unwrap();
+
+    if remaining==1 {
+        me.yield_with((my_number, 0));
+    } else {
+        let mut coros: Vec<coroutine::asymmetric::Coroutine<(u64, u64)>> = Vec::with_capacity(10);
+
+        for _ in 0..10 {
+            coros.push(Coroutine::spawn(&skynet));
```

Comment:
More idiomatic yet practically equivalent is

    let coros: Vec<_> = (0..10).map(|_| Coroutine::spawn(&skynet)).collect();